lib/workers/evm_server.rb

```diff
@@ -214,4 +262,43 @@ def start_workers
     @server.sync_workers
     @server.wait_for_started_workers
   end
+
+  ######################################################################
+  # Warning:
+  #
+  # The following methods can lead to unexpected (and likely unpleasant)
+  # behavior if used out of the scope of the orchestrator process.
+  #
+  # They change the global state which is used to determine the current
+  # server's identity. This intentionally will alter the values of calls
+  # such as MiqServer.my_server and MiqServer.my_guid, and also the
+  # contents of the global ::Settings constant.
+  ######################################################################
+  def for_each_server
```

Comment:
`for_each_server` the method name feels too generic for me.  Perhaps `as_each_server`, or `each_impersonated_server`?